website/cue/reference/components.cue

```diff
@@ -49,6 +49,15 @@ components: {
 
 		configuration: #Schema
 
+		// This field should be used in cases where auto-generating example configs using fields
+		// like `common` and `required` breaks down. If no `example_config` is set on a component,
+		// the example config is auto-generated; if `example_config` is set, then this map will be
+		// used to generate the example directly.
+		example_config?: {
+			common:    #Fields
+			advanced?: #Fields
+		}
```

Comment:
Could we make this just take an arbitrary key/value list? I'm imagining a world where a component could have 3, 4, or even more examples. I think, with the move to providing hand-written examples, that we can also move away from the "Common" / "Advanced" terminology which is tied to the current example generation which uses the `common` flag on the config options.

Consider, for example, the examples at the bottom of the route doc: https://deploy-preview-9870--vector-project.netlify.app/docs/reference/configuration/transforms/route/#examples

I think we could even just remove that section and pull up those examples?

👍 to making this an additive change so that we can migrate components over time.